osu.Game/Storyboards/Storyboard.cs

```diff
@@ -90,8 +91,8 @@ public bool ReplacesBackground
             }
         }
 
-        public DrawableStoryboard CreateDrawable(IWorkingBeatmap working = null) =>
-            new DrawableStoryboard(this);
+        public DrawableStoryboard CreateDrawable(IReadOnlyList<Mod> mods = null) =>
+            new DrawableStoryboard(this, mods);
```

Comment:
What's with the nullability choices here? `GameplayState.Mods` is initially guaranteed to be non-null, and so is `OsuTestScene.SelectedMods.Value`. So why does this suddenly allow null, which then `DrawableStoryboard` internally *fixes up* to non-null again by the `?? Array.Empty<Mod>` fallback?

Also why is the param optional if all usages pass it in anyway?

Reply:
Was going with how `DrawableRuleset` does it:
https://github.com/ppy/osu/blob/e67b1fe0ec9c50b7ddb429edb2ee13d534e881ad/osu.Game/Rulesets/Ruleset.cs#L192-L198
https://github.com/ppy/osu/blob/53c901bfa8051bab3ff96aee65ec8c16472764b3/osu.Game/Rulesets/UI/DrawableRuleset.cs#L116-L126

And you can't make it non-null without making it non-optional, but since there aren't many usages to this, I suppose it's fine making it non-optional.